packages/flutter/test/material/badge_test.dart

```diff
@@ -438,4 +438,34 @@ void main() {
     // Badge should scale with content
     expect(box, paints..rrect(rrect: RRect.fromLTRBR(0, -7, 30 + 8, 23, badgeRadius)));
   });
+
+  testWidgets('Badge.count maxCount limits displayed value', (WidgetTester tester) async {
+    Widget buildFrame(int count, [int maxCount = 999]) {
+      return MaterialApp(
+        home: Align(
+          alignment: Alignment.topLeft,
+          child: Badge.count(count: count, maxCount: maxCount, child: const Icon(Icons.add)),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(buildFrame(5, 99));
```

Comment:
Add Edge Case Test for Negative Count
Even though it’s unlikely to happen in production, explicitly asserting that negative count values throw an AssertionError or are clamped would increase robustness.